src/opower/utilities/bge.py

```diff
@@ -21,3 +28,145 @@ def subdomain() -> str:
+                    "tx": settings["transId"],
+                    "p": settings["hosts"]["policy"],
+                    "diags": json.dumps(
+                        {
+                            "pageViewId": settings["pageViewId"],
+                            "pageId": settings["api"],
+                            "trace": [],
+                        }
+                    ),
+                },
+                headers={"User-Agent": USER_AGENT},
+                raise_for_status=True,
+            ) as resp:
+                result = await resp.text(encoding="utf-8")
+
+            async with session.get(
+                "https://"
+                + cls.login_domain()
+                + "/api/Services/MyAccountService.svc/GetSession",
+                headers={"User-Agent": USER_AGENT},
+                raise_for_status=True,
+            ) as resp:
+                result = await resp.json()
+
+            if result["accountNumber"] is None:
+                bearer_token = result["token"]
+                # if we don't yet have an account, look one up and set it
+                # this path comes from GetConfiguration:
+                # "euApiRoutePrefix": "/mobile/custom",
+                eu_api_route_prefix = "/mobile/custom"
+                # "euApiUrl": "/.euapi",
+                eu_api_url = "/.euapi"
+                async with session.get(
+                    "https://"
+                    + cls.login_domain()
+                    + eu_api_url
+                    + eu_api_route_prefix
+                    + "/auth/accounts",
+                    headers={
+                        "User-Agent": USER_AGENT,
+                        "Authorization": "Bearer " + bearer_token,
+                    },
+                    raise_for_status=True,
+                ) as resp:
+                    # this has the wrong mime type for some reason
+                    result = await resp.json(content_type="text/html")
+
+                if result["success"] is not True:
+                    raise InvalidAuth("Unable to list accounts")
+
+                active_accounts = [
+                    account
+                    for account in result["data"]
+                    if account["status"] == "Active"
+                ]
+                if len(active_accounts) == 0:
+                    raise InvalidAuth("No active accounts found")
+
+                # set the first active one
+                account_number = active_accounts[0]["accountNumber"]
+
+                async with session.post(
+                    "https://"
+                    + cls.login_domain()
+                    + "/api/Services/AccountList.svc/ViewAccount",
+                    json={
+                        "accountNumber": account_number,
+                    },
+                    headers={"User-Agent": USER_AGENT},
+                    raise_for_status=True,
+                ) as resp:
+                    result = await resp.text(encoding="utf-8")

```

Comment:
While you are here, can you fix `DEBUG:charset_normalizer:Encoding detection: utf_8 is most likely the one.` that I get even without your changes when running with `-v`? I guess we need to specify this for the other resp.text calls in the file. I did fix that warning in an earlier commit but I guess the 2nd login I added in the demo caused another such a warning.